examples/platforms/emsk/radio.c

```diff
@@ -0,0 +1,660 @@
+static bool sIsReceiverEnabled = false;
+
+static volatile uint8_t Mrf24StatusTx = 0;
+static volatile uint8_t Mrf24StatusRx = 0;
+static volatile uint8_t Mrf24StatusSec = 0;
+
+static DEV_SPI_PTR pmrf_spi_ptr;
+static DEV_GPIO_PTR pmrf_gpio_ptr;
+static void RadioIsr(void *ptr);
+
+/* Variables for test */
+static uint32_t numInterruptRev = 0;
+static uint32_t numInterruptTrans = 0;
+static uint32_t numRadioProcess = 0;
+static uint32_t NODE_ID = 1;
+
+static inline bool isSecurityEnabled(const uint8_t *frame)
+{
+    return (frame[0] & IEEE802154_SECURITY_ENABLED) != 0;
+}
+
+static inline bool isAckRequested(const uint8_t *frame)
+{
+    return (frame[0] & IEEE802154_ACK_REQUEST) != 0;
+}
+
+static inline bool isPanIdCompressed(const uint8_t *frame)
+{
+    return (frame[0] & IEEE802154_PANID_COMPRESSION) != 0;
+}
+
+static inline uint8_t getHeadLength(const uint8_t *frame)
+{
+    uint8_t length = 0;
+    /* Frame Control-2 + Sequence Number-1 */
+    length += 2 + 1;
+
+    /* Destination PAN + Address */
+    switch (frame[1] & IEEE802154_DST_ADDR_MASK)
+    {
+        case IEEE802154_DST_ADDR_SHORT:
+            length += sizeof(otPanId) + sizeof(otShortAddress);
+            break;
+
+        case IEEE802154_DST_ADDR_EXT:
+            length += sizeof(otPanId) + sizeof(otExtAddress);
+            break;
+
+        default:
+            ExitNow(length = 0);
+    }
+
+    /* Source PAN + Address */
+    switch (frame[1] & IEEE802154_SRC_ADDR_MASK)
+    {
+        case IEEE802154_SRC_ADDR_SHORT:
+            if (!isPanIdCompressed(frame))
+            {
+                length += sizeof(otPanId);
+            }
+
+            length += sizeof(otShortAddress);
+            break;
+
+        case IEEE802154_SRC_ADDR_EXT:
+            if (!isPanIdCompressed(frame))
+            {
+                length += sizeof(otPanId);
+            }
+
+            length += sizeof(otExtAddress);
+            break;
+
+        default:
+        ExitNow(length = 0);
+    }
+
+exit:
+    return length;
+}
+
+void enableReceiver(void)
+{
+    if (!sIsReceiverEnabled)
+    {
+        mrf24j40_rxfifo_flush();
+        /* add code to enable receiver (wakeup) */
+        sIsReceiverEnabled = true;
+    }
+}
+
+void disableReceiver(void)
+{
+    if (sIsReceiverEnabled)
+    {
+        mrf24j40_rxfifo_flush();
+        /* add code to disable receiver (sleep) */
+        sIsReceiverEnabled = false;
+    }
+}
+
+void setChannel(uint8_t channel)
+{
+    mrf24j40_set_channel((int16_t)(channel - 11));
+}
+
+void otPlatRadioGetIeeeEui64(otInstance *aInstance, uint8_t *aIeeeEui64)
+{
+    (void)aInstance;
+    /* should set it manually or preset it in memory */
+    aIeeeEui64[0] = 0x18;
+    aIeeeEui64[1] = 0xb4;
+    aIeeeEui64[2] = 0x30;
+    aIeeeEui64[3] = 0x00;
+    aIeeeEui64[4] = (NODE_ID >> 24) & 0xff;
+    aIeeeEui64[5] = (NODE_ID >> 16) & 0xff;
+    aIeeeEui64[6] = (NODE_ID >> 8) & 0xff;
+    aIeeeEui64[7] = NODE_ID & 0xff;
```

Comment:
Thanks.
EMSK is a Xilinx FPGA-based hardware platform for ARC processor. Please refer to 
https://www.synopsys.com/dw/ipdir.php?ds=arc_em_starter_kit for details.
 
There is no wireless support in EMSK. So We use Microchip MRF24J40 as radio module, without EUI support. A additional EEPROM is required for EUI. I am trying to find a way to get a unique ID. 
I will update it soon.